src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/core/RcaConf.java

```diff
@@ -202,17 +203,24 @@ public CacheDeciderConfig getCacheDeciderConfig() {
     return setting;
   }
 
-  public void updateAllRcaConfFiles(final Set<String> mutedRcas, final Set<String> mutedDeciders,
+  public boolean updateAllRcaConfFiles(final Set<String> mutedRcas, final Set<String> mutedDeciders,
       final Set<String> mutedActions) {
+    boolean updateStatus = true;
     // update all rca.conf files
     List<String> rcaConfFiles = RcaControllerHelper.getAllConfFilePaths();
-
     for (String confFilePath : rcaConfFiles) {
-      updateRcaConf(confFilePath, mutedRcas, mutedDeciders, mutedActions);
+      updateStatus = updateRcaConf(confFilePath, mutedRcas, mutedDeciders,
+        mutedActions);
+      if (!updateStatus) {
+        LOG.error("Failed to update the conf file at path: {}", confFilePath);
+        StatsCollector.instance().logMetric(RcaConsts.WRITE_UPDATED_RCA_CONF_ERROR);
```

Comment:
You can use the new metric framework. There your metric can also tell you which file erred out ?